contracts/ethereum/contracts/Token.sol

```diff
@@ -165,21 +165,33 @@ contract Token is
     /// @notice burnFee_ that's paid by the user when they mint
     uint256 private mintFee_;
 
+    /* ~~~~~~~~~~ VOLUME TRACKNG ~~~~~~~~~~ */
+
+    struct EpochVolume {
```

Comment:
Can you put the struct outside the scope of this and into an interface? And add documentation explaining what it does? And can you not use a mapping inside the struct?